### ISSUE TITLE
Python 3.4 (Keras): It seems the kernel died unexpectedly. Use 'Restart kernel' to continue using this console

### ISSUE THREAD

**Issue by nadith** [spyder-ide/spyder#3426]
## Description of your problem
Spyder IPython kernel display "It seems the kernel died unexpectedly. Use 'Restart kernel' to continue using this console."
        
   class VGGNet16:

    def __init__(self, weights_path=None):
        self.model = Sequential()

        self.model.add(ZeroPadding2D((1,1),input_shape=(3,224,224)))
        self.model.add(Convolution2D(64, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(64, 3, 3, activation='relu'))
        self.model.add(MaxPooling2D((2,2), strides=(2,2)))

        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(128, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(128, 3, 3, activation='relu'))
        self.model.add(MaxPooling2D((2,2), strides=(2,2)))
    
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(256, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(256, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(256, 3, 3, activation='relu'))
        self.model.add(MaxPooling2D((2,2), strides=(2,2)))
    
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(512, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(512, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(512, 3, 3, activation='relu'))
        self.model.add(MaxPooling2D((2,2), strides=(2,2)))
    
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(512, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(512, 3, 3, activation='relu'))
        self.model.add(ZeroPadding2D((1,1)))
        self.model.add(Convolution2D(512, 3, 3, activation='relu'))
        self.model.add(MaxPooling2D((2,2), strides=(2,2)))
    
        self.model.add(Flatten())
        self.model.add(Dense(4096, activation='relu'))
        self.model.add(Dropout(0.5))
        self.model.add(Dense(4096, activation='relu'))
        self.model.add(Dropout(0.5))
        self.model.add(Dense(1000, activation='softmax'))
    
        if weights_path:
            self.model.load_weights(weights_path)
        
        sgd = SGD(lr=0.1, decay=1e-6, momentum=0.9, nesterov=True)
        self.model.compile(optimizer=sgd, loss='categorical_crossentropy')

        self.get_layer_output = K.function([self.model.layers[0].input, K.learning_phase()],
                                  [self.model.layers[34].output])        
        
    def Test(self, X_Tr):
        # output in test mode = 0
        out = self.get_layer_output([X_Tr, 0])[0]        
        return out;

**What steps will reproduce the problem?**

1. Initialize a object from the above class
2. Run Test(..) method for few times (feed forward 1 image at a time), X_tr = [1, 3, 224, 224]
3. spyder IPython console will display "It seems the kernel died unexpectedly. Use 'Restart kernel' to continue using this console."
I also tried Ipython qtconsole. Same problem exists.

**What is the expected output? What do you see instead?**
Expected output is the 4096 feature from the VGG net. Instead IPython console crashed after running the Test method few times.

**Please provide any additional information below**
PC Spec:
Processor: Intel 5820k
RAM: 16GB
HardDisk: 1TB (50GB for page file)
GPUs: Titan X (Pascal), GTX 970

## Versions and main components
* Spyder Version: 2.3.9
* Python Version: 3.4.5
* Operating system: Windows 10 x64

## Dependencies
IPython: 4.2.0

Spyder Optional Dependencies:
IPython >=1.0      :  4.2.0 (OK)
jedi >=0.8.1;<0.9.0:  0.9.0 (NOK)
matplotlib >=1.0   :  1.5.3 (OK)
pandas >=0.13.1    :  0.18.1 (OK)
pep8 >=0.6         :  1.7.0 (OK)
pyflakes >=0.6.0   :  1.2.3 (OK)
pygments >=1.6     :  2.1.3 (OK)
pylint >=0.25      :  None (NOK)
qtconsole >=4.0    :  4.2.1 (OK)
rope >=0.9.2       :  0.9.4-1 (OK)
sphinx >=0.6.6     :  1.4.1 (OK)
sympy >=0.7.3      :  1.0 (OK)
zmq >=2.1.11       :  15.4.0 (OK)

Please help me out since I tried many solutions that exist in various forums but nothing worked.
*conda install pyzmq=14.7.0 did not work too

Thank you.

Related Issue: #3114

**Comment by nadith** [comment]
I figure out where the issue was by myself. Im posting the info for others.
It was not about Spyder or IPython console. The error was about the BLAS implementation openBLAS built which was not compatible with the other dependent libs such as NumPY.

I ran the script with python in the prompt: and I observed 
 ** On entry to SGEMM  parameter number  3 had an illegal value
which is infact an issue with the openblas lib that I have linked.

**Comment by SystemErrorWang** [comment]
Hello nadith, I came across the same problem working on 64bit windows10 system, but my keras have tensorflow backend. Do you have any idea about it? Thank you so much.

**Comment by sdoyen** [comment]
Thanks Nadith.
I was facing a similar issue where Keras would cause the kernel to die when loading a model.
I am sure there must a better way but I was able to clear this issue by refactoring my code and making sure Keras was loaded before numpy.
